ms2query/sqlite.py

```diff
@@ -0,0 +1,136 @@
+        File location of the json file that stores the spectra.
+    table_name:
+        Name of the table in the database to which the spectra should be added,
+        default = "spectra"
+    """
+    spectra = json.load(open(json_spectrum_file_name))
+    conn = sqlite3.connect(sqlite_file_name)
+
+    # Get the column names in the sqlite file
+    cur = conn.execute(f'select * from {table_name}')
+    column_names = list(map(lambda x: x[0], cur.description))
+
+    # Add the information of each spectrum to the sqlite file
+    for spectrum in spectra:
+        columns = ""
+        values = ""
+        # Check if there is a key for the spectrum that is the same as the
+        # specified columns and if this is the case add this value to this
+        # column.
+        for column in column_names:
+            if column in spectrum:
+                # Add comma when it is not the first column that is added
+                if len(columns) > 0:
+                    columns += ", "
+                    values += ", "
+                columns += column
+                values += '"' + spectrum[column] + '"'
+            else:
+                print(f"No value found for column: {column} in "
+                      f"spectrum {spectrum['spectrum_id']}")
+        # Add the complete spectrum in json format to the column "full_json"
+        if "full_json" in column_names:
+            columns += ", full_json"
+            values += f', "{spectrum}" '
+        add_spectrum_command = f"INSERT INTO {table_name} " \
+                               + f"({columns}) values ({values})"
+
+        cur = conn.cursor()
+        cur.execute(add_spectrum_command)
+        conn.commit()
+    conn.close()
+
+
+def get_spectra_from_sqlite(sqlite_file_name: str,
+                            spectrum_id_list: List[str],
+                            table_name: str = "spectra") -> List[dict]:
+    """Returns a list with all metadata of spectrum_ids in spectrum_id_list
+
+    Args:
+    -------
+    sqlite_file_name:
+        File name of the sqlite file that contains the spectrum information
+    spectrum_id_list:
+        List of spectrum_id's of which the metadata should be returned
+    table_name:
+        Name of the table in the sqlite file that stores the spectrum data
+
+    Returns:
+    -------
+    sqlite_spectra:
+    """
+    conn = sqlite3.connect(sqlite_file_name)
+
+    sqlite_command = f"""SELECT full_json FROM {table_name} 
+                    WHERE spectrum_id 
+                    IN ('{"', '".join(map(str, spectrum_id_list))}')"""
+
+    cur = conn.cursor()
+    cur.execute(sqlite_command)
+
+    sqlite_spectra = []
+    for json_spectrum in cur:
+        # Remove the "()" around the spectrum
+        json_spectrum = json_spectrum[0]
+        # Convert string to dictionary
+        json_spectrum = ast.literal_eval(json_spectrum)
+        sqlite_spectra.append(json_spectrum)
+    conn.close()
+
+    return sqlite_spectra
```

Comment:
Maybe better to first convert it to matchms type `Spectrum` objects.

Reply:
or here: https://github.com/iomega/ms2query/blob/main/ms2query/utils.py

Reply:
The matchms function was perfect for converting from dict to Spectrum

Reply:
Nice!